src/openforms/forms/api/serializers/logic/action_serializers.py

```diff
@@ -151,10 +163,14 @@ def validate(self, data: dict) -> dict:
         if (
             action_type
             and action_type == LogicActionTypes.step_not_applicable
-            and not form_step
+            and (not form_step and not form_step_uuid)
         ):
             raise serializers.ValidationError(
-                {"form_step": self.fields["form_step"].error_messages["null"]},
+                {
+                    "form_step_uuid": self.fields["form_step_uuid"].error_messages[
+                        "null"
+                    ]
+                },
```

Comment:
Probably if `form_step` is provided but `form_step_uuid` isn't, we need to convert the former into the latter in the data (for backwards compatibility reasons)

edit: I see that `src/openforms/submissions/logic/actions.py` handles both old and new situation